### PR TITLE
Support segment zone map

### PR DESCRIPTION
**Pull request by liutang123**
#1860  Support segment zone map and key range.

### CODE REVIEW

**Comment by liutang123** [comment]
cc @gaodayue support segment zone map.

**Review by gaodayue — CHANGES_REQUESTED** [review]

**Comment by gaodayue** [comment]
Also note that `BetaRowset`, `Segment`, and `ColumnReader` should be thread safe but the current implementation is not. However I think this should be fixed in a separate PR.

File: gensrc/proto/segment_v2.proto [hunk 1]
```diff
@@ -91,8 +91,10 @@ message ColumnMetaPB {
     optional bool is_nullable = 6;
     // ordinal index page
     optional PagePointerPB ordinal_index_page = 7;
-    // zone map page
+    // zone map page of zone maps in page granularity
     optional PagePointerPB zone_map_page = 8;
+    // zone map in segment granularity
+    optional ZoneMapPB zone_map = 10;
```
[the remark this answers is not in the document]
Comment (liutang123) [reply gensrc/proto/segment_v2.proto]: `zone_map` is a field of `ColumnMetaPB`. Is `column_zone_map ` really better than `zone_map`?

[the remark this answers is not in the document]
Comment (liutang123) [reply gensrc/proto/segment_v2.proto]: id 9 was occupied by dict_page.

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 2]
```diff
@@ -50,17 +52,69 @@ Status Segment::open() {
...
+            int32_t column_id = column_condition.first;
+            auto entry = _column_id_to_footer_ordinal.find(column_id);
+            if (entry == _column_id_to_footer_ordinal.end()) {
+                continue;
+            }
+            auto& c_meta = _footer.columns(entry->second);
+            if (!c_meta.has_zone_map()) {
+                continue;
+            }
+            ZoneMapPB c_zone_map = c_meta.zone_map();
```
Comment (imay) [inline be/src/olap/rowset/segment_v2/segment.cpp]:
```suggestion
            auto& c_zone_map = c_meta.zone_map();
```

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**